### PR TITLE
Run typecheck on the SuperSpec and fix signature in builtin.qnt 

### PR DESCRIPTION
**Pull request by shonfeder**
This changeset was initially meant to fix https://github.com/informalsystems/quint/pull/617/commits/3370abe9d524ff7051c68267dae493daa3ccbc9a but along the way I encountered a few bits of tech debt and stale/untested stuff. Please see individual commits for details.

- [x] Tests added for any new code
- [x] Ran `npm run format` (or had formatting run automatically on all files edited)
- [x] Documentation added for any new functionality
- [x] Entries added to the respective `CHANGELOG.md` for any new functionality
- [x] Feature table on [`README.md`](../README.md#roadmap) updated for any listed functionality
- [x] Regenerated documentation with `quint docs` if `builtin.qnt` was changed

### CODE REVIEW

**Review by konnov — APPROVED** [review]
That's really cool! Thanks for fixing all these issues!

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -5,4 +5,4 @@
 - [ ] Documentation added for any new functionality
 - [ ] Entries added to the respective `CHANGELOG.md` for any new functionality
 - [ ] Feature table on [`README.md`](../README.md#roadmap) updated for any listed functionality
-- [ ] Regenerated documentation with `quint docs` if `builtin.qnt` was changed
```
Comment (bugarela) [inline .github/pull_request_template.md:8]: Do we actually need a checklist item for this if it is already being checked by CI?

**Review by bugarela — APPROVED** [review]
Thanks!

File: .github/pull_request_template.md (hunk 1 above)
Comment (shonfeder) [reply .github/pull_request_template.md:8]: Makes sense.

**Pull request merged**